pkg/kubelet/metrics/metrics.go

```diff
@@ -134,6 +135,24 @@ var (
 			StabilityLevel: metrics.ALPHA,
 		},
 	)
+	// PodStartSLIDuration is a Histogram that tracks the duration (in seconds) it takes for a single pod to run,
+	// excluding the time for image pulling. This metric should reflect the "Pod startup latency SLI" definition
+	// ref: https://github.com/kubernetes/community/blob/master/sig-scalability/slos/pod_startup_latency.md
+	//
+	// The histogram bucket boundaries for pod startup latency metrics, measured in seconds. These are hand-picked
+	// so as to be roughly exponential but still round numbers in everyday units. This is to minimise the number
+	// of buckets while allowing accurate measurement of thresholds which might be used in SLOs
+	// e.g. x% of pods start up within 30 seconds, or 15 minutes, etc.
+	PodStartSLIDuration = metrics.NewHistogramVec(
+		&metrics.HistogramOpts{
+			Subsystem:      KubeletSubsystem,
+			Name:           PodStartSLIDurationKey,
+			Help:           "Duration in seconds to start a pod, without image pull times",
```

Comment:
I think it would help to be a bit more precise about what "start a pod" means. For example, kubelet can start a pod, but then the status update to the API server that the pod was started can be delayed (e.g. due to QPS limits of kubelet). 

I would be a bit more specific around what "start" means here. Is it just until kubelet observes a pod is in running phase and image is pulled? Or the API update to API server was posted? (Based on your implementation it seems like the latter (since the update is in status_manager).

Reply:
Agree. I've used the exact wording from "Pod startup latency SLI/SLO Definition"